pkg/metadata/host/host.go

```diff
@@ -150,7 +151,16 @@ func getHostAliases() []string {
 		aliases = append(aliases, tencentAlias)
 	}
 
-	return aliases
+	validatedAliases := []string{}
+	for _, alias := range util.SortUniqInPlace(aliases) {
+		if err := validate.ValidHostname(alias); err == nil {
```

Comment:
Did you check all host aliases are valid hostnames for all providers?

Reply:
All external providers appear to return aliases that are valid hostnames, but you have a good point. It would be safer to apply validation only to the aliases configured by the user.